src/Processors/Merges/Algorithms/RowRef.h

```diff
@@ -129,14 +129,19 @@ struct RowRef
     size_t num_columns = 0;
     UInt64 row_num = 0;
 
+    int source_stream_index = -1;
```

Comment:
std::optional?

Reply:
Yes, I think we can just use `UInt64`, because when `hasEqualSortColumnsWith` is called the object surely has been initialised.